core/services/ocr2/plugins/generic/telemetry_adapter.go

```diff
@@ -6,33 +6,36 @@ import (
 
 	"github.com/smartcontractkit/libocr/commontypes"
 
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/services/telemetry"
+
 	"github.com/smartcontractkit/chainlink-relay/pkg/types"
 )
 
 var _ types.TelemetryService = (*TelemetryAdapter)(nil)
 
 type TelemetryAdapter struct {
-	endpointGenerator types.MonitoringEndpointGenerator
+	endpointGenerator telemetry.MonitoringEndpointGenerator
 	endpoints         map[[4]string]commontypes.MonitoringEndpoint
 }
 
-func NewTelemetryAdapter(endpointGen types.MonitoringEndpointGenerator) *TelemetryAdapter {
+func NewTelemetryAdapter(endpointGen telemetry.MonitoringEndpointGenerator) *TelemetryAdapter {
 	return &TelemetryAdapter{
 		endpoints:         make(map[[4]string]commontypes.MonitoringEndpoint),
 		endpointGenerator: endpointGen,
 	}
 }
 
 func (t *TelemetryAdapter) Send(ctx context.Context, network string, chainID string, contractID string, telemetryType string, payload []byte) error {
-	e, err := t.getOrCreateEndpoint(contractID, telemetryType, network, chainID)
+	e, err := t.getOrCreateEndpoint(network, chainID, contractID, telemetryType)
 	if err != nil {
 		return err
 	}
 	e.SendLog(payload)
 	return nil
 }
 
-func (t *TelemetryAdapter) getOrCreateEndpoint(contractID string, telemetryType string, network string, chainID string) (commontypes.MonitoringEndpoint, error) {
+func (t *TelemetryAdapter) getOrCreateEndpoint(network string, chainID string, contractID string, telemetryType string) (commontypes.MonitoringEndpoint, error) {
```

Comment:
perhaps out of scope for this change, but should the telemetry API be changed to take the relayer.ID struct (which is the network and chain id) rather than the individual components?

Reply:
Let me think about this and potentially address this in the follow up I mentioned to Jordan: https://smartcontract-it.atlassian.net/jira/software/c/projects/BCF/boards/49?assignee=712020%3A25361daa-6eb8-4a03-9963-0264ac70a192&selectedIssue=BCF-2750